internal/monzo/monzo_client.go

```diff
@@ -0,0 +1,245 @@
+// Package monzo handles transaction events and webhooks for the Monzo banking service
+package monzo
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/baely/txn/internal/common/errors"
+)
+
+const monzoBaseURI = "https://api.monzo.com"
+
+// MonzoClient handles API interactions with Monzo
+type MonzoClient struct {
+	accessToken string
+	client      *http.Client
+}
+
+// NewMonzoClient creates a new client for the Monzo API
+func NewMonzoClient(accessToken string) *MonzoClient {
+	return &MonzoClient{
+		accessToken: accessToken,
+		client: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
+
+// request makes an API request to the Monzo API
+func (c *MonzoClient) request(ctx context.Context, method, endpoint string, payload []byte, ret interface{}) error {
+	var body *bytes.Buffer
+	if payload != nil {
+		body = bytes.NewBuffer(payload)
+	} else {
+		body = bytes.NewBuffer(nil)
+	}
+
+	uri := fmt.Sprintf("%s/%s", monzoBaseURI, endpoint)
```

Comment:
Sprintf is relatively expensive to run per request for  just basic string concat

Reply:
Addressed in commit 39e43bf by replacing fmt.Sprintf with direct string concatenation.

Reply:
Fixed in commit 39e43bf - all instances of fmt.Sprintf for simple string concatenation have been replaced with direct string addition.

Reply:
Fixed in commit 39e43bf by replacing fmt.Sprintf with direct string concatenation for improved performance.